modules/azure_devops_agent_container_app_jobs/variables.tf

```diff
@@ -0,0 +1,94 @@
+#######################################
+# Azure resources specific variables  #
+#######################################
+variable "resource_group_name" {
+  type        = string
+  description = "Name of the resource group that the resources should be placed in. Check for naming conflicts."
+}
+
+variable "resource_prefix" {
+  type        = string
+  description = "Prefix for resources"
+  validation {
+    condition     = length(var.resource_prefix) < 12 && var.resource_prefix == lower(var.resource_prefix)
+    error_message = "resource_prefix must be 11 chars or less and consist of only lower case alphanumerical characters."
+  }
+}
+
+variable "kv_ip_rules" {
+  default     = []
+  type        = set(string)
+  description = "IPs that will be allowed to access the KV holding the secrets needed by the environment"
+  validation {
+    condition = alltrue([
+      for ip in var.kv_ip_rules : can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", ip))
+    ])
+    error_message = "Invalid IP address provided in kv_ip_rules"
+  }
+}
+
+variable "infrastructure_subnet_id" {
+  type        = string
+  description = "The subnet_id where the container app jobs are running. The Subnet must have a /21 or larger address space."
+}
+
+variable "additional_tags" {
+  type        = map(string)
+  description = "Additional tags that should be added to all resources. Concated with the default tags"
+  default     = {}
+}
+
+#######################################
+# Github tied variables               #
+#######################################
+variable "azp_token" {
+  sensitive   = true
+  type        = string
+  description = "Base64 encoded Azure Devops PAT"
+}
+
+variable "azp_org_url" {
+  sensitive   = true
+  type        = string
+  description = "URL for your Azure DevOps organization"
+}
+
+#######################################
+# Runner tied variables               #
+#######################################
+
+variable "agent_image" {
+  type        = string
+  default     = "ghcr.io/tjololo/container-apps-ci-cd-runner-tutorial:latest"
+  description = "Docker image to run when a job is scheduled"
+}
+
+variable "agent_cpu" {
+  type        = string
+  default     = "0.5"
+  description = "CPU allocated to a runner"
+}
+
+variable "agent_memory" {
+  type        = string
+  default     = "1Gi"
+  description = "Memory allocated to a runner"
+}
+
+variable "agent_pool_name" {
+  type        = string
+  default     = "demo-azure-containers"
```

Comment:
I'm not super familiar with the setup but I suppose a default value might not make sense here? Or it should default to some other convention being used somewhere else?

Reply:
nvm, the downward API would not work here: https://github.com/Altinn/terraform-azurerm-altinn-modules/pull/6/files#diff-18114ba9c0d286fb0649b0969c953a23368de8e9568b5d8a5e7fef61b2a6f8e0R124

Reply:
not a kubernetes node pool, but a azure devops agent pool. Default makes no sense as you say

Reply:
Yeah I belive it's an k8s cluster down there somewhere, with keda and other services running. I don't think you can specify any actual nodepool for container apps as Azure owns and manages the clusters running the workloads. You can specify how much cpu and memory you want to reserve (and pay for), that's about it.
This variable, named by me in the terraform modules variables, references the azure devops term [agent pool](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=azure-devops&tabs=yaml%2Cbrowser).